### DISCUSSION TITLE
Is this project alive?  If so, a proposal.

### DISCUSSION THREAD

**Discussion by fsiler** [RaRe-Technologies/smart_open#698] [Ideas]
I'm wondering about the possibility of refactoring the scheme to instead maybe be a tuple that allows the specification of a regex.  The reason is that I'd like to be able to make `smart_open` directly understand URLs from cloud providers, such as `*.blob.core.windows.net` and the corresponding URLs for other providers.  Is this something of interest that might stand a chance of getting merged?

**Comment by mpenkov** [comment]
Yes, the project is alive (we just did a release a few weeks ago). I think your proposal is out of scope - you're welcome to either fork smart_open or build another layer of logic on top of it. If it were me, I'd go with the latter, but you most definitely understand your own use case better than I do.
